### PR TITLE
[feat]: [CDS-58192]: Return resolved yaml in inputset api

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
## Describe your changes
We want pipelines/execution/{planExecutionId}/inputsetV2 and pipelines/execution/{planExecutionId}/inputset to return inputYaml where only trigger expressions are resolved. So we have added new parameter(enum type) based on which we return a yaml where all the expressions are resolved or only trigger expressions are resolved. 
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46003)

### CODE REVIEW

File: pipeline-service/service/src/main/java/io/harness/pms/helpers/YamlExpressionResolveHelper.java [hunk 1]
```diff
@@ -70,6 +73,33 @@ public String resolveExpressionsInYaml(String yamlString, String planExecutionId
     throw new InvalidRequestException("Invalid request : No execution details found");
   }
 
+  public String resolveTriggerExpressionsInYaml(String yamlString, String planExecutionId) {
```
Comment (sribalij) [inline pipeline-service/service/src/main/java/io/harness/pms/helpers/YamlExpressionResolveHelper.java:76]: This method will still resolve all other expressions and not just `<+trigger.>` expressions right? Is this not similar to `resolveExpressionsInYaml(..)`

**Review by sribalij — DISMISSED** [review]
> Reviewed 4 of 9 files at r1, 2 of 3 files at r2, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46003)* status: 9 file(s) to review. 1 of 2 approvals obtained.

**Review by sribalij — DISMISSED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46003)* status: 9 file(s) to review. 1 of 2 approvals obtained.

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 3 of 9 files at r1, 2 of 3 files at r2, 3 of 3 files at r4, 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46003)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Comment by rathodmeetsatish** [comment]
trigger ut2

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/NGTriggerExpressionEvaluator.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import io.harness.annotations.dev.HarnessTeam;
+import io.harness.annotations.dev.OwnedBy;
+import io.harness.engine.executions.plan.PlanExecutionMetadataService;
+import io.harness.expression.EngineExpressionEvaluator;
+import io.harness.ngtriggers.expressions.functors.TriggerFunctor;
+import io.harness.pms.contracts.ambiance.Ambiance;
+import io.harness.pms.plan.execution.SetupAbstractionKeys;
+
+@OwnedBy(HarnessTeam.SPG)
+public class NGTriggerExpressionEvaluator extends EngineExpressionEvaluator {
```
Comment (sribalij) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/NGTriggerExpressionEvaluator.java:19]: Looks like we already have `TriggerExpressionEvaluator` for triggers but more targeted toward webhook triggers. That already resolves `<+trigger.payload. >` `<+eventPayload>` expressions. 
We can at a later point see if we can combine then together. 
cc: @vcalasansh

Comment (sribalij) [reply pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/NGTriggerExpressionEvaluator.java:19]: them*

**Comment by rathodmeetsatish** [comment]
trigger ut0,trigger ut1,trigger ut2,trigger ut3,trigger ut4,trigger ut5,trigger ut6,trigger ut7,trigger ut8,trigger ut9

**Comment by rathodmeetsatish** [comment]
trigger ut6,trigger ut0

**Comment by rathodmeetsatish** [comment]
trigger review

**Review by sribalij — APPROVED** [review]
> Reviewed 2 of 3 files at r6, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46003)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46003)* status: :shipit: complete! ✓

**Comment by rathodmeetsatish** [comment]
trigger review

**Pull request merged**